File: backend/laravel_api/app/Models/Postulacion.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Postulacion extends Model
{
    use HasFactory;

    protected $table = 'postulaciones';

    protected $fillable = [
        'vacante_id',
        'user_id',
        'nombre',
        'cv_path',
        'mensaje',
        'estado',
    ];

    public function vacante()
    {
        return $this->belongsTo(Vacante::class, 'vacante_id');
    }

    public function usuario()
    {
        return $this->belongsTo(User::class, 'user_id');
    }

    public function postulaciones()
    {
    return $this->hasMany(Postulacion::class, 'vacante_id');
    }

}

File: intrahub_reports/app/models/postulacion.rb
class Postulacion < ApplicationRecord
  self.table_name = "postulaciones"   # Usa la tabla creada por Laravel

  belongs_to :vacante, class_name: "Vacante", foreign_key: "vacante_id", optional: true
  belongs_to :user, class_name: "User", foreign_key: "user_id", optional: true
end
